### PR TITLE
Update versions to 1.0.1-alpha for all packages.

### PR DESCRIPTION
**Pull request by MaxKsyunz**
Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

### Description
Bump version to 1.0.1 and set it to pre-release.
This allows all packages to be updated on nuget.org and require users to acknowledge that these are not GA packages.

Once the packages and process are validated, -alpha suffix can be removed to make the packages GA.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Yury-Fridlyand — APPROVED** [review]

File: Directory.Build.props [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <Project>
   <!-- Default Version numbers -->
   <PropertyGroup>
-    <CurrentVersion>1.0.0</CurrentVersion>
-    <CurrentAssemblyVersion>1.0.0</CurrentAssemblyVersion>
-    <CurrentAssemblyFileVersion>1.0.0</CurrentAssemblyFileVersion>
+    <CurrentVersion>1.0.1-prerelease</CurrentVersion>
```
Comment (prudhvigodithi) [inline Directory.Build.props:4]: Hey @MaxKsyunz from description does this has to be 1.0.1-alpha?

File: USER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+This user guide specifies how to include and use the .NET client in your application.
+
+### Getting Started
+
+Include OSC in your .csproj file.
+```xml
+<Project>
+  ...
+  <ItemGroup>
+    <PackageReference Include="Osc" Version="1.0.0" />
```
Comment (a-alamoudi) [inline USER_GUIDE.md:22]: Maybe `<PackageReference Include="opensearch.client" Version="1.0.0" />`?

File: src/OpenSearch.Client/Document/Multiple/BulkAll/BulkAllRequest.cs [hunk 3]
```diff
@@ -78,7 +78,7 @@ public interface IBulkAllRequest<T> where T : class
 		///<summary>The index to use for items that don't specify one. By default, will be inferred from <typeparamref name="T"/>.
 		/// If no default index has been mapped for <typeparamref name="T" />
 		/// using <see cref="ConnectionSettingsBase{TConnectionSettings}.DefaultMappingFor{TDocument}"/>
-		/// on <see cref="Osc.ConnectionSettings"/>, an exception will be thrown.
+		/// on <see cref="ConnectionSettings"/>, an exception will be thrown.
```
Comment (acarbonetto) [inline src/OpenSearch.Client/Document/Multiple/BulkAll/BulkAllRequest.cs:81]: cref="Client. ConnectionSettings"

**Review by acarbonetto — APPROVED** [review]

File: USER_GUIDE.md (hunk 2 above)
Comment (MaxKsyunz) [reply USER_GUIDE.md]: Thank you. Addressed in 8ddc23a2be05d799ebaa96f6ec557553174aab4e

File: src/OpenSearch.Client/Document/Multiple/BulkAll/BulkAllRequest.cs (hunk 3 above)
Comment (MaxKsyunz) [reply src/OpenSearch.Client/Document/Multiple/BulkAll/BulkAllRequest.cs:81]: That references resolves correctly as is -- both BulkAllRequest and ConnectionSettings are in the same namespace -- OpenSearch.Client.

**Pull request merged**